src/services/state-persistor/utils/FeatureStyleHelper.ts

```diff
@@ -0,0 +1,783 @@
+    return showLabel
+  }
+
+  public getSizeProperty(feature: Feature) {
+    return this.getNumber(feature, luxFormatFeatureProperties.SIZE)
+  }
+
+  public getStrokeProperty(feature: Feature) {
+    return this.getNumber(feature, luxFormatFeatureProperties.STROKE)
+  }
+
+  // === OTHER UTILITY METHODS ===
+
+  public createTextStyle_(options: { [key: string]: any }) {
+    if (options.angle) {
+      const angle = options.angle !== undefined ? options.angle : 0
+      const rotation = (angle * Math.PI) / 180
+      options.rotation = rotation
+      delete options.angle
+    }
+
+    options.font = ['normal', `${options.size || 10}pt`, 'Arial'].join(' ')
+
+    if (options.color) {
+      options.fill = new olStyleFill({ color: options.color || [0, 0, 0, 1] })
+      delete options.color
+    }
+
+    options.stroke = new olStyleStroke({
+      color: [255, 255, 255, 1],
+      width: options.width || 3,
+    })
+    delete options.width
+
+    return new olStyleText(options)
+  }
+
+  /**
+   * Get the measure of the given feature as a string. For points, you can format
+   * the result by setting a filter to apply on the coordinate with the function
+   * {@link ngeo.misc.FeatureHelper.prototype.setPointFilterFn}.
+   */
+  public getMeasure(feature: Feature) {
+    const geometry = feature.getGeometry()
+    console.assert(geometry, 'Geometry should be truthy')
+
+    let measure = ''
+
+    if (geometry instanceof olGeomPolygon) {
+      if (this.getType(feature) === styleGeometryType.CIRCLE) {
+        const azimut = this.optNumber(
+          feature,
+          luxFormatFeatureProperties.AZIMUT
+        )
+        console.assert(typeof azimut === 'number')
+        const line = this.getRadiusLine(feature, azimut || 10)
+
+        measure = getFormattedAzimutRadius(
+          line,
+          this.projection_!,
+          this.decimals_,
+          this.precision_ || 3
+        )
+      } else {
+        measure = getFormattedArea(geometry) //, this.projection_, this.precision_, this.unitPrefixFormat_);
+      }
+    } else if (geometry instanceof olGeomLineString) {
+      measure = getFormattedLength(geometry, this.projection_!) //, this.precision_, this.unitPrefixFormat_);
+    } else if (geometry instanceof olGeomPoint) {
+      if (this.pointFilterFn_ === null) {
+        measure = getFormattedPoint(geometry, this.decimals_)
+      } else {
+        const coordinates = geometry.getCoordinates()
+        const args = this.pointFilterArgs_.slice(0)
+        args.unshift(coordinates)
+        measure = (this.pointFilterFn_ || ((c: boolean) => (c ? '' : '')))(
+          ...args
+        )
+      }
+    }
+
+    return measure
+  }
+
+  /**
+   * Return the type of geometry of a feature using its geometry property and
+   * some inner properties.
+   */
+  public getType(feature: Feature) {
+    const geometry = feature.getGeometry()
+    console.assert(geometry, 'Geometry should be thruthy')
```

Comment:
why use console.assert in v4?

Reply:
does is matter ?

it was a goog assert and I wanted to preserve the verification

Reply:
yes code quality matters